### ISSUE TITLE
Cannot correctly resume

### ISSUE THREAD

**Issue by yyNoBug** [tyshiwo1/DiM-DiffusionMamba#10]
Dear authors! Thank you for your active response for maintaining this codebase. I recently realized that the codebase cannot be properly resumed. 

I've noticed that the `lr_scheduler` and `nn_ema` have not been resumed in your codebase, so I added these two lines 
```python
lr_scheduler.load_state_dict(torch.load(os.path.join(ckpt_path, 'lr_scheduler.pth'), map_location='cpu'))
nnet_ema.load_state_dict(torch.load(os.path.join(ckpt_path, 'nnet_ema.pth'), map_location='cpu'))
```
However, the experiment still cannot be properly resumed.

Please see the following image. When the experiment resumes at checkpoint 100k, the loss curve actually is not continuous from previous run. Have you noticed this problem before?

<img width="513" alt="Screenshot 2024-08-11 at 3 33 11 AM" src="https://github.com/user-attachments/assets/f47585bc-15ef-4a53-9262-36e23588c6c7">

**Comment by yyNoBug** [comment]
Not the bug of the original codebase.
